src/providers/ipa/ipa_deskprofile_rules_util.c

```diff
@@ -0,0 +1,796 @@
+                                         char **_rule_name)
+{
+    TALLOC_CTX *tmp_ctx;
+    char buffer[1024];
+    size_t buffer_len;
+    size_t name_len;
+
+    tmp_ctx = talloc_new(mem_ctx);
+    if (tmp_ctx == NULL) {
+        return ENOMEM;
+    }
+
+    name_len = strlen(name);
+    buffer_len = 0;
+    for (size_t i = 0; i < name_len; i++) {
+        char character;
+        bool replace;
+
+        character = name[i];
+        replace = false;
+
+        if (isalnum(character) == 0) {
+            char next_character;
+
+            next_character = name[i+1];
+            if (i + 1 >= name_len || isalnum(next_character) == 0) {
+                continue;
+            }
+
+            replace = true;
+        }
+
+        buffer[buffer_len] = replace ? '_' : character;
+        buffer_len++;
+    }
+    buffer[buffer_len] = '\0';
+
+    *_rule_name = talloc_strdup(mem_ctx, buffer);
+    return EOK;
+}
+
+static errno_t
+ipa_deskprofile_rule_check_memberuser(
+                                TALLOC_CTX *mem_ctx,
+                                struct sysdb_attrs *rule,
+                                const char *rule_name,
+                                const char *rule_prio,
+                                const char *base_dn,
+                                const char *username, /* non fully-qualified */
+                                char **_user_prio,
+                                char **_group_prio)
+{
+    TALLOC_CTX *tmp_ctx;
+    struct ldb_message_element *el;
+    char *data;
+    char *memberuser;
+    char *membergroup;
+    char *user_prio;
+    char *group_prio;
+    bool user = false;
+    bool group = false;
+    errno_t ret;
+
+    tmp_ctx = talloc_new(mem_ctx);
+    if (tmp_ctx == NULL) {
+        return ENOMEM;
+    }
+
+    ret = sysdb_attrs_get_el(rule, IPA_MEMBER_USER, &el);
+    if (ret != EOK) {
+        DEBUG(SSSDBG_TRACE_FUNC,
+              "Failed to get the Desktop Profile Rule memberUser for rule "
+              "\"%s\" [%d]: %s\n",
+              rule_name, ret, sss_strerror(ret));
+
+        goto done;
+    }
+
+    memberuser = talloc_asprintf(tmp_ctx, "uid=%s,cn=users,cn=accounts,%s",
+                                 username, base_dn);
+    if (memberuser == NULL) {
+        DEBUG(SSSDBG_CRIT_FAILURE, "Failed to allocate memberuser\n");
+        ret = ENOMEM;
+        goto done;
+    }
+
+    membergroup = talloc_asprintf(tmp_ctx, "cn=groups,cn=accounts,%s", base_dn);
+    if (membergroup == NULL) {
+        DEBUG(SSSDBG_CRIT_FAILURE, "Failed to allocate membergroup\n");
+        ret = ENOMEM;
+        goto done;
+    }
+
+    for (size_t i = 0; i < el->num_values; i++) {
+        if (user && group) {
+            break;
+        }
+
+        data = (char *)el->values[i].data;
+        if (!user && data != NULL && strcmp(memberuser, data) == 0) {
+            user = true;
+            continue;
+        }
+
+        if (!group && data != NULL && strstr(data, membergroup) != NULL) {
```

Comment:
Yep, this part was completely wrong.
I've ended up adding a proper handler based on HBAC's code (and sharing the code whenever was possible/made sense).

Fixed locally!